src/sql/workbench/services/notebook/sql/sqlSessionManager.ts

```diff
@@ -225,12 +228,14 @@ class SqlKernel extends Disposable implements nb.IKernel {
 			}
 			this._queryRunner.runQuery(code);
 		} else if (this._currentConnection) {
-			let connectionUri = Utils.generateUri(this._currentConnection, 'notebook');
-			this._queryRunner = this._instantiationService.createInstance(QueryRunner, connectionUri);
-			this._connectionManagementService.connect(this._currentConnection, connectionUri).then((result) => {
+			let connectionUri = this._connectionManagementService.getConnectionUriFromId(this._currentConnectionProfile.id);
+			if (!this._connectionManagementService.isConnected(connectionUri)) {
```

Comment:
So what's the experience like for this? Here's the scenario I'm imagining :

1. User connects to a server in OE
2. They create a new notebook with a connection to that server
3. They then disconnect from the server in OE

Is the Notebook still going to be runnable, or does it create a separate connection when the Notebook is created that's separate from the OE one? If it's not going to be runnable is there a way to reconnect from the Notebook (I would imagine it'd be like the query window where it'll automatically connect you if needed)